### PR TITLE
Test results api

### PR DESCRIPTION
**Pull request by Prevole**
Not ready yet!

### CODE REVIEW

**Comment by Prevole** [comment]
Added the filter by test on the project versions API. The API tests is quite slow but enforce the scoping of data.

File: app/api/probe_dock/project_versions_api.rb [hunk 1]
```diff
@@ -26,6 +26,8 @@ def current_organization
             Organization.active.where(normalized_name: params[:organizationName].to_s.downcase).first!
           elsif params[:projectId].present?
             Organization.active.joins(:projects).where('projects.api_id = ?', params[:projectId].to_s).first!
+          elsif params[:testId].present?
+            ProjectTest.where(api_id: params[:testId]).first!.project.organization
```
Comment (AlphaHydrae) [inline app/api/probe_dock/project_versions_api.rb]: This makes 2 queries and loads data that will not be used. The relation should start from Organization and make the necessary joins like the other queries in this method. It also allows you to use `Organization.active` which makes sure that it only works for active organizations. Please also update `ResultsApi#current_organization` to use this pattern.

File: app/api/probe_dock/results_api.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            ProjectTest.where(api_id: params[:testId]).first!.project.organization
+          end
+        end
+      end
+
+      get do
+        authorize! TestResult, :index
+
+        rel = TestResult
+          .order('test_results.active desc, test_results.passed, test_results.name, test_results.id')
```
Comment (AlphaHydrae) [inline app/api/probe_dock/results_api.rb]: This order clause was meant for display to display the results in test reports, where we display failing results first, then passing ones, then inactive ones.

In the case of the test widget, we'll want to order by run date.

I would leave this as the default order, but allow the user to give a `sort=runAt` param which will change the order.

Comment (AlphaHydrae) [inline app/api/probe_dock/results_api.rb]: This order clause was meant for display to display the results in test reports, where we display failing results first, then passing ones, then inactive ones.

In the case of the test widget, we'll want to order by run date.

I would leave this as the default order, but allow the user to give a `sort=runAt` param which will change the order.

File: client/pages/report-details/details.reportResults.directive.js [hunk 3]
```diff
@@ -81,7 +81,7 @@ angular.module('probedock.reportDetailsPage').directive('reportDetailsResults',
       $scope.fetchingMoreResults = true;
 
       return api({
-        url: '/reports/' + $scope.report.id + '/results',
+        url: '/results?reportId=' + $scope.report.id,
```
Comment (AlphaHydrae) [inline client/pages/report-details/details.reportResults.directive.js]: Please pass the `reportId` query parameter in the `params` object.

File: spec/support/named_records_spec_helper.rb [hunk 4]
```diff
@@ -16,15 +16,55 @@
...
+  end
+
+  def named_record_by_type(name, type)
+    key = NamedRecordKey.new(name, type)
+    record = @named_records.try(:[], key)
+    raise "Unknown named record #{key}" unless record
     record
   end
+
+  class NamedRecordKey
```
Comment (AlphaHydrae) [inline spec/support/named_records_spec_helper.rb]: Remove if unused.

File: app/api/probe_dock/project_versions_api.rb (hunk 1 above)
Comment (Prevole) [inline app/api/probe_dock/project_versions_api.rb]: Done

File: app/api/probe_dock/results_api.rb (hunk 2 above)
Comment (Prevole) [inline app/api/probe_dock/results_api.rb]: Done

File: client/pages/report-details/details.reportResults.directive.js (hunk 3 above)
Comment (Prevole) [inline client/pages/report-details/details.reportResults.directive.js]: Done

File: spec/support/named_records_spec_helper.rb (hunk 4 above)
Comment (Prevole) [inline spec/support/named_records_spec_helper.rb]: Done (kept in GIT history :))

**Pull request merged**